packages/admin-scripts/src/application/admin-script-base.js

```diff
@@ -0,0 +1,138 @@
+const { createScriptExecutionRepository } = require('@friggframework/core/admin-scripts/repositories/script-execution-repository-factory');
+const { createAdminApiKeyRepository } = require('@friggframework/core/admin-scripts/repositories/admin-api-key-repository-factory');
+
+/**
+ * Admin Script Base Class
+ *
+ * Base class for all admin scripts. Provides:
+ * - Standard script definition pattern
+ * - Repository access
+ * - Logging helpers
+ * - Integration factory support (optional)
+ *
+ * Usage:
+ * ```javascript
+ * class MyScript extends AdminScriptBase {
+ *   static Definition = {
+ *     name: 'my-script',
+ *     version: '1.0.0',
+ *     description: 'Does something useful',
+ *     ...
+ *   };
+ *
+ *   async execute(frigg, params) {
+ *     // Your script logic here
+ *   }
+ * }
+ * ```
+ */
+class AdminScriptBase {
+    /**
+     * CHILDREN SHOULD SPECIFY A DEFINITION FOR THE SCRIPT
+     * Pattern matches IntegrationBase.Definition
+     */
+    static Definition = {
+        name: 'Script Name', // Required: unique identifier
+        version: '0.0.0', // Required: semver for migrations
+        description: 'What this script does', // Required: human-readable
+
+        // Script-specific properties
+        source: 'USER_DEFINED', // 'BUILTIN' | 'USER_DEFINED'
+
+        inputSchema: null, // Optional: JSON Schema for params
+        outputSchema: null, // Optional: JSON Schema for results
```

Comment:
what's the idea with these schemas?

Reply:
Validation, and reporting. Largely validation.